sw/device/lib/dif/dif_flash_ctrl.h

```diff
@@ -11,13 +11,1096 @@
+   * truncate to the closest, lower word aligned address. For example, if 0x13
+   * is supplied for a read, the controller will perform a read at address 0x10.
+   *
+   * Similarly, for page erase and bank erase operations, the flash controller
+   * will truncate to the closest lower page-aligned or bank-aligned address,
+   * respectively.
+   */
+  uint32_t byte_address;
+  /**
+   * Number of 32-bit words to program. Must be in the range [1,4096]
+   * (inclusive).
+   */
+  uint32_t word_count;
+} dif_flash_ctrl_transaction_t;
+
+/**
+ * Start a flash controller operation.
+ *
+ * Note that not all underlying flash memory supports the
+ * `kDifFlashCtrlOpProgramRepair` operation. When specifying this in the `op`,
+ * `dif_flash_ctrl_query_repair_support()` can first be used to ensure the
+ * operation is supported. If a repair operation is requested and not supported
+ * by the underlying hardware, the `kDifBadArg` error will be returned from this
+ * function.
+ *
+ * @param handle The flash controller device to execute the transaction on.
+ * @param transaction The parameters that constitute the operation to start.
+ * @return `kDifBadArg` if `handle` is null or `transaction` contains an invalid
+ * or unsupported operation, `kDifUnavailable` if a flash transaction is in
+ * progress at the controller, `kDifIpFifoFull` if the FIFOs are not empty, and
+ * `kDifOk` otherwise.
+ */
+OT_WARN_UNUSED_RESULT
+dif_result_t dif_flash_ctrl_start(dif_flash_ctrl_state_t *handle,
+                                  dif_flash_ctrl_transaction_t transaction);
+
+/**
+ * Start a flash controller operation, with fewer safeguards. (unsafe version)
+ *
+ * Unlike `dif_flash_ctrl_start()`, this variant does not check whether the info
+ * partition id is valid, whether the word count is valid, or whether the FIFOs
+ * have been emptied. This function is intended to be used to trigger errors on
+ * purpose, for testing the hardware responses.
+ *
+ * Note that not all underlying flash memory supports the
+ * `kDifFlashCtrlOpProgramRepair` operation. When specifying this in the `op`,
+ * `dif_flash_ctrl_query_repair_support()` can first be used to ensure the
+ * operation is supported. If a repair operation is requested and not supported
+ * by the underlying hardware, the `kDifBadArg` error will be returned from this
+ * function.
+ *
+ * @param handle The flash controller device to execute the transaction on.
+ * @param transaction The parameters that constitute the operation to start.
+ * @return `kDifBadArg` if `handle` is null, `kDifUnavailable` if a flash
+ * transaction is in progress, and `kDifOk` otherwise.
+ */
+OT_WARN_UNUSED_RESULT
+dif_result_t dif_flash_ctrl_start_unsafe(
+    dif_flash_ctrl_state_t *handle, dif_flash_ctrl_transaction_t transaction);
+
+/**
+ * Suspend an ongoing erase transaction.
+ *
+ * This function blocks until hardware reports that the erase suspension
```

Comment:
Whoops, this isn't true anymore. I decided to avoid forcing a blocking call here. Instead, the caller can poll `dif_flash_ctrl_get_erase_suspend_status()` and spin on it. Need to update the comment.